crypto/evp/digest.c

```diff
@@ -180,32 +180,31 @@ static int evp_md_init_internal(EVP_MD_CTX *ctx, const EVP_MD *type,
      * previous handle, re-querying for an ENGINE, and having a
      * reinitialisation, when it may all be unnecessary.
      */
-    if (ctx->engine && ctx->digest &&
-        (type == NULL || (type->type == ctx->digest->type)))
+    if (ctx->engine && ctx->digest && type->type == ctx->digest->type)
```

Comment:
Would you please add != NULL for the first two conditions?

Reply:
Done.